packages/cli/src/index.ts

```diff
@@ -12,183 +12,285 @@ import {
 
 import type {
   FarmCLIBuildOptions,
+  FarmCLICommonOptions,
   FarmCLIPreviewOptions,
   FarmCLIServerOptions,
   GlobalFarmCLIOptions,
-  ICleanOptions
+  ICleanOptions,
+  NormalizedFarmCLIBuildOptions
 } from './types.js';
 
 const { version } = JSON.parse(
   readFileSync(new URL('../package.json', import.meta.url)).toString()
 );
 
-const cli = cac('farm');
-
-// common command
-cli
-  .option('-c, --config <file>', 'use specified config file')
-  .option('-m, --mode <mode>', 'set env mode')
-  .option('--base <path>', 'public base path')
-  .option('--clearScreen', 'allow/disable clear screen when logging', {
-    default: true
-  });
-
-// dev command
-cli
-  .command(
-    '[root]',
-    'Compile the project in dev mode and serve it with farm dev server'
-  )
-  .alias('start')
-  .alias('dev')
-  .option('-l, --lazy', 'lazyCompilation')
-  .option('--host <host>', 'specify host')
-  .option('--port <port>', 'specify port')
-  .option('--open', 'open browser on server start')
-  .option('--hmr', 'enable hot module replacement')
-  .option('--cors', 'enable cors')
-  .option('--strictPort', 'specified port is already in use, exit with error')
-  .action(
-    async (
-      rootPath: string,
-      options: FarmCLIServerOptions & GlobalFarmCLIOptions
-    ) => {
-      const { root, configPath } = resolveCliConfig(rootPath, options);
-      const resolveOptions = resolveCommandOptions(options);
-
-      const defaultOptions = {
-        root,
-        compilation: {
-          lazyCompilation: options.lazy
-        },
-        server: resolveOptions,
-        clearScreen: options.clearScreen,
-        configPath,
-        mode: options.mode
-      };
-
-      const { start } = await resolveCore();
-      handleAsyncOperationErrors(
-        start(defaultOptions),
-        'Failed to start server'
-      );
+const devCommand = defineCommand({
```

Comment:
The original idea was to cut files with all commands, start command, watch build preview, create new command, create different files in the folder and write them.

Reply:
OK, I'll refactor this later